### PR TITLE
Feature local storage

### PR DESCRIPTION
**Pull request by BlesslinJerishR**
- Add LocalStorage functionality for the form

### CODE REVIEW

**Review by Qoosim — COMMENTED** [review]
#  *Status: Invalid Request* :red_circle: 

Hi @team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.



## Invalid Code Review Request

You have submitted a project, that 

- [ ] has commits only from one group member in a pari-programming project. Every member must have at least one commit in the PR.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: script.js [hunk 1]
```diff
@@ -12,3 +12,83 @@ function clickX() {
...
+
+// const ul = document.getElementById('badges');
+// const skills = ['Ruby on Rails', 'CSS', 'JS'];
+// const li = document.createElement('li');
+// for (const skill of skills) {
+//   const text = document.createTextNode(skill);
+//   li.appendChild(text);
+//   ul.appendChild(li);
+// }
+// li.classList.add('badge');
```
Comment (Ibizugbe) [inline script.js]: Kindly take away every commented block of code as they make your code base look a bit messy and less professional😉

File: script.js [hunk 2]
```diff
@@ -12,3 +12,83 @@ function clickX() {
...
+  if (errorMessages.length > 0) {
+    event.preventDefault();
+    error.textContent = errorMessages.join('.');
+  } else {
+    error.textContent = '';
+  }
+  localStorage.setItem('name', name.value);
+  localStorage.setItem('email', email.value);
+  localStorage.setItem('text', txt.value);
+});
```
Comment (Ibizugbe) [inline script.js]: Nice work so far👍. Your contact form information is not retained when you reload the page🥲 and this is a requirement in this milestone. kindly make the necessary change👍

image

File: script.js [hunk 3]
```diff
@@ -12,3 +12,83 @@ function clickX() {
...
+    errorMessages.push('Email must be in lowercase');
+  }
+  if (errorMessages.length > 0) {
+    event.preventDefault();
+    error.textContent = errorMessages.join('.');
+  } else {
+    error.textContent = '';
+  }
+  localStorage.setItem('name', name.value);
+  localStorage.setItem('email', email.value);
```
Comment (Ibizugbe) [inline script.js]: It is required that You must create a single JavaScript object with all the data from the entire form and save it in local storage. kindly implement this feature to retain data from your local storage on your input fields on page load 👍

image

kindly ensure the features in the[ requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/10_localstorage.md) are implemented👍

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @BlesslinJerishR 👋👋, 

tryagain


Good job so far!👍
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] The [JavaScript](https://github.com/microverseinc/linters-config/tree/master/javascript) and [HTML&CSS](https://github.com/microverseinc/linters-config/tree/master/html-css) linters is set correctly project repository.The [JavaScript](https://github.com/microverseinc/linters-config/tree/master/javascript) and [HTML&CSS](https://github.com/microverseinc/linters-config/tree/master/html-css) linters are correctly setup and they are passing👍.
- [x] Github flow practice has been used👍. 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by amejid — COMMENTED** [review]
Hi team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, where linter checks are not running. Please resolve the merge conflict so linters can run properly.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Strangeal — APPROVED** [review]
Hi @BlesslinJerishR, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Alt Text

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**